paasta_tools/mesos_maintenance.py

```diff
@@ -334,6 +379,118 @@ def load_credentials(mesos_secrets='/nail/etc/mesos-slave-secret'):
     return username, password
 
 
+def get_principal(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the principal from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the principal/username
+    """
+    return load_credentials(mesos_secrets)[0]
+
+
+def get_secret(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the secret from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the secret/password
+    """
+    return load_credentials(mesos_secrets)[1]
+
+
+def reserve(slave_id, resource, amount):
+    """Dynamically reserve resources in marathon to prevent tasks from using them.
+    :param slave_id: the id of the mesos slave
+    :param resource: the mesos resource type we want to reserve (i.e. cpus, mem, gpus, etc)
+    :param amount: integer representing the amount of the above resource to reserve
+    :returns: boolean where 0 represents success and 1 is a failure
+    """
+    log.info("Dynamically reserving %f %s on %s" % (amount, resource, slave_id))
+    payload = {
+        'slaveId': slave_id,
+        'resources': str(build_reservation_payload(slave_id, resource, amount)).replace("'", '"').replace('+', '%20')
```

Comment:
@Rob-Johnson That is how I initially had the code. Apparently, while Mesos wants the data submitted as `application/x-www-form-urlencoded`, it parses it like JSON. As a result, it complains if you have single quotes instead of double quotes. `requests` also encodes whitespace as `+`, while Mesos is expecting `%20`.

@mattmb I can try messing around with `urllib.quote`. I hate these `replace()` calls just as much as both of you.